### ISSUE TITLE
nat can end up in `round-robin` state

### ISSUE THREAD

**Issue by igalic** [ployground/bsdploy#102]
When configuring a jailhost on digitalocean, its network may end up looking like this
```
root@meenix:~ # pfctl -s nat
nat on vtnet0 inet from 10.0.0.2 to any -> (vtnet0) round-robin
…
```

This will cause the jails' networking to be an order of magnitude slower, than on the body. I don't know what this is caused by; it could have something to do with digitalocean's default private network:

```
root@meenix:~ # ifconfig
vtnet0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
        options=6c07bb<RXCSUM,TXCSUM,VLAN_MTU,VLAN_HWTAGGING,JUMBO_MTU,VLAN_HWCSUM,TSO4,TSO6,LRO,VLAN_HWTSO,LINKSTATE,RXCSUM_IPV6,TXCSUM_IPV6>
        ether 3a:0c:c6:47:2f:12
        inet6 fe80::380c:c6ff:fe47:2f12%vtnet0 prefixlen 64 scopeid 0x1 
        inet6 2a03:b0c0:3:d0::3585:b001 prefixlen 64 
        inet 207.154.239.110 netmask 0xfffff000 broadcast 207.154.239.255 
        inet 10.19.0.6 netmask 0xffff0000 broadcast 10.19.255.255 
        nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
        media: Ethernet 10Gbase-T <full-duplex>
        status: active
vtnet1: flags=8802<BROADCAST,SIMPLEX,MULTICAST> metric 0 mtu 1500
        options=6c07bb<RXCSUM,TXCSUM,VLAN_MTU,VLAN_HWTAGGING,JUMBO_MTU,VLAN_HWCSUM,TSO4,TSO6,LRO,VLAN_HWTSO,LINKSTATE,RXCSUM_IPV6,TXCSUM_IPV6>
        ether 1e:4f:85:54:53:2c
        nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
        media: Ethernet 10Gbase-T <full-duplex>
        status: active
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> metric 0 mtu 16384
        options=600003<RXCSUM,TXCSUM,RXCSUM_IPV6,TXCSUM_IPV6>
        inet6 ::1 prefixlen 128 
        inet6 fe80::1%lo0 prefixlen 64 scopeid 0x3 
        inet 127.0.0.1 netmask 0xff000000 
        nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
        groups: lo 
lo1: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> metric 0 mtu 16384
        options=600003<RXCSUM,TXCSUM,RXCSUM_IPV6,TXCSUM_IPV6>
        inet 10.0.0.2 netmask 0xffffffff 
        nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
        groups: lo 
root@meenix:~ # netstat -nr
Routing tables

Internet:
Destination        Gateway            Flags     Netif Expire
default            207.154.224.1      UGS      vtnet0
10.0.0.2           link#4             UH          lo1
10.19.0.0/16       link#1             U        vtnet0
10.19.0.6          link#1             UHS         lo0
127.0.0.1          link#3             UH          lo0
207.154.224.0/20   link#1             U        vtnet0
207.154.239.110    link#1             UHS         lo0

Internet6:
Destination                       Gateway                       Flags     Netif Expire
::/96                             ::1                           UGRS        lo0
default                           2a03:b0c0:3:d0::1             UGS      vtnet0
::1                               link#3                        UH          lo0
::ffff:0.0.0.0/96                 ::1                           UGRS        lo0
2a03:b0c0:3:d0::/64               link#1                        U        vtnet0
2a03:b0c0:3:d0::3585:b001         link#1                        UHS         lo0
fe80::/10                         ::1                           UGRS        lo0
fe80::%vtnet0/64                  link#1                        U        vtnet0
fe80::380c:c6ff:fe47:2f12%vtnet0  link#1                        UHS         lo0
fe80::%lo0/64                     link#3                        U           lo0
fe80::1%lo0                       link#3                        UHS         lo0
ff02::/16                         ::1                           UGRS        lo0
root@meenix:~ #
```

i changed the `nat` rule to `nat on vtnet0 from lo1:network to any -> (vtnet0:0)` and this made it worked!

i don't know if changing the [pf.conf](https://github.com/ployground/bsdploy/blob/master/bsdploy/roles/jails_host/templates/pf.conf) template to:

```mustache
{% for network in pf_nat_jail_networks %}
nat on {{ pf_nat_interface }} from {{ network }} to any -> ({{ pf_nat_interface }}:0)
{% endfor %}
```

would make it work for everyone.

**Comment by fschulze** [comment]
I've tried it on my NAS and it seems to work fine. Will test further.

**Comment by fschulze** [comment]
FTR, the command to see if round-robin is active is ``pfctl -s nat``

**Comment by igalic** [comment]
yesterday i speculated that maybe `ezjail-admin start` would be the one that times out, and hence prevents `ploy start <instance>` to break off
however, after today's test with multi-ip / ipv6: https://github.com/ployground/ploy_ezjail/issues/10 and seeing that `ploy start test` failed (with a long stack-trace) but that `ezjail-admin create` was actually successful, i started the jail in question only to find that it takes, roughly, forever to complete (install python)

This is now, *despite* that fact that `pf` looks still the same:
```pf
nat on vtnet0 inet from 10.0.0.0/8 to any -> (vtnet0:0)
pass in all flags S/SA keep state
pass out all flags S/SA keep state
```

IPv6 routes are… messy:

```
root@meenirx:~ # netstat -nr
Routing tables

Internet:
Destination        Gateway            Flags     Netif Expire
default            139.59.144.1       UGS      vtnet0
10.0.0.2           link#4             UH          lo1
10.0.0.3           link#4             UH          lo1
10.19.0.0/16       link#1             U        vtnet0
10.19.0.5          link#1             UHS         lo0
10.135.0.0/16      link#2             U        vtnet1
10.135.4.227       link#2             UHS         lo0
127.0.0.1          link#3             UH          lo0
139.59.144.0/20    link#1             U        vtnet0
139.59.155.43      link#1             UHS         lo0

Internet6:
Destination                       Gateway                       Flags     Netif Expire
::/96                             ::1                           UGRS        lo0
default                           2a03:b0c0:3:d0::1             UGS      vtnet0
::1                               link#3                        UH          lo0
::ffff:0.0.0.0/96                 ::1                           UGRS        lo0
2a03:b0c0:3:d0::/64               link#1                        U        vtnet0
2a03:b0c0:3:d0::3a4d:c001         link#1                        UHS         lo0 # why is this bound to lo0?
2a03:b0c0:3:d0::3a4d:c002         link#4                        UHS         lo0 # in addition to this?
fe80::/10                         ::1                           UGRS        lo0
fe80::%vtnet0/64                  link#1                        U        vtnet0
fe80::ccbf:c4ff:fe1b:42ca%vtnet0  link#1                        UHS         lo0
fe80::%vtnet1/64                  link#2                        U        vtnet1
fe80::9c9c:3aff:fed3:2c6d%vtnet1  link#2                        UHS         lo0
fe80::%lo0/64                     link#3                        U           lo0
fe80::1%lo0                       link#3                        UHS         lo0
ff02::/16                         ::1                           UGRS        lo0
root@meenirx:~ #
```

in the jail, `traceroute` works fine:
```
root@test:~ # traceroute google.com
traceroute to google.com (172.217.19.110), 64 hops max, 40 byte packets
 1  139.59.144.253 (139.59.144.253)  0.801 ms
    139.59.144.254 (139.59.144.254)  0.490 ms  0.711 ms
 2  138.197.250.142 (138.197.250.142)  3.428 ms
    138.197.250.156 (138.197.250.156)  0.512 ms
    138.197.250.152 (138.197.250.152)  0.590 ms
 3  138.197.250.140 (138.197.250.140)  2.297 ms  1.025 ms
    138.197.250.134 (138.197.250.134)  0.487 ms
 4  74.125.146.56 (74.125.146.56)  0.520 ms
    216.239.47.98 (216.239.47.98)  0.768 ms
    74.125.146.56 (74.125.146.56)  0.553 ms
 5  216.239.47.48 (216.239.47.48)  1.560 ms
    209.85.247.83 (209.85.247.83)  0.820 ms
    216.239.57.188 (216.239.57.188)  5.481 ms
 6  209.85.247.83 (209.85.247.83)  0.663 ms
    216.239.40.7 (216.239.40.7)  7.450 ms
    66.249.95.153 (66.249.95.153)  7.112 ms
 7  216.239.40.7 (216.239.40.7)  16.222 ms
    216.239.51.59 (216.239.51.59)  8.057 ms  7.658 ms
 8  209.85.240.93 (209.85.240.93)  7.705 ms  7.698 ms  8.202 ms
 9  209.85.240.93 (209.85.240.93)  8.047 ms
    muc03s07-in-f14.1e100.net (172.217.19.110)  7.131 ms
    209.85.240.93 (209.85.240.93)  7.709 ms
root@test:~ #
```
while `traceroute6` does not:
```
root@test:~ # traceroute6 google.com
traceroute6 to google.com (2a00:1450:4016:802::200e) from 2a03:b0c0:3:d0::3a4d:c002, 64 hops max, 12 byte packets
 1  * * *
 2  * * *
 3  * * *
 4  * * *
 5  * * *
 6  * * *
 7  * * *
 8  * * *
 9  * * *
10  * * *
11  * * *
12  * * *
13  * * *
```
this still doesn't quite explain the awful performance of `pkg update` & `pkg install` — Doesn't FreeBSD implement [Happy Eyeballs](https://en.wikipedia.org/wiki/Happy_Eyeballs)?

**Comment by fschulze** [comment]
It seems like this also affects servers with multiple IP4 adresses. @tomster this seems to fix our backup server issue from today.

**Comment by fschulze** [comment]
Released with 2.3.0
